api/organisations/permissions/permissions.py

```diff
@@ -90,3 +93,17 @@ def has_object_permission(self, request, view, obj):
             return True
 
         return False
+
+
+class NestedIsOrganisationAdminPermission(BasePermission):
+    def has_permission(self, request, view):
+        organisation_pk = view.kwargs.get("organisation_pk")
+
+        with suppress(ObjectDoesNotExist):
+            return request.user.is_organisation_admin(
+                Organisation.objects.get(pk=organisation_pk)
+            )
+        return False
+
+    def has_object_permission(self, request, view, obj):
+        return self.has_permission(request, view)
```

Comment:
Hmm... I'd prefer to grab the organisation from the object itself here. The view kwargs can be applied incorrectly and you'd possibly end up getting authorised where you shouldn't.

Reply:
You mean like in a scenario where we override get_object or something to essentially return a different organisation? 
wouldn't that be a bad design decision?
PS: yeah, since it's permissions it does make sense to err on the side of caution

Reply:
Although with that applied it becomes less generic and can only be applied to object that have direct relationship to Organisation with an attribute named `organisation`

Reply:
Yep, Done